xdsl/ir/core.py

```diff
@@ -947,6 +947,17 @@ def get_traits_of_type(cls, trait_type: type[OpTraitInvT]) -> list[OpTraitInvT]:
         """
         return [t for t in cls.traits if isinstance(t, trait_type)]
 
+    def get_parent_with_trait(
```

Comment:
I would want to call this `get_ancestor_with_trait`, but I'm not sure if we use `parent` elsewhere for the same concept. Have you found another funciton in MLIR or xDSL where parent is not the direct parent op?

Reply:
+1

Reply:
The name is copied from the MLIR codebase. There’s `getParentOfType()` and `getParentWithTrait()` that go up symbols using `getParentOp()`. Then there are `isAncestor()` and `isProperAncestor()` that go up symbols using `getParentRegion()`.

Reply:
Ok when in doubt we stick with MLIR conventions even when they don't make a lot of sense :)